### PR TITLE
[Merged by Bors] - make miner produce proposal

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation

5th PR for #2902, miner produce proposal

## Changes

- make miner produce proposal and convert back to Block
- move eligibility oracle to miner package

## Test Plan

unit test

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by countvonzero** [comment]
bors try

File: cmd/node/node.go [hunk 1]
```diff
@@ -600,28 +597,25 @@ func (app *App) initServices(ctx context.Context,
...
+		atxDB,
 		app.host,
-		clock.Subscribe(),
 		msh,
 		trtl,
-		blockOracle,
 		tBeacon,
 		newSyncer,
-		stateAndMeshProjector,
+		pendingtxs.NewStateAndMeshProjector(processor, msh).GetProjection,
```
Comment (dshulyak) [inline cmd/node/node.go]: using interface has a benefit that you can find all implementations of the interface with a single command in IDE's

File: miner/oracle.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	}
+
+	logger.With().Info("proposal eligibility calculated",
+		log.Uint32("total_num_slots", numEligibleSlots),
+		log.Int("num_layers_eligible", len(eligibilityProofs)),
+		log.String("layers_to_num_proposals", prettyPrintProofs(eligibilityProofs)))
+	return eligibilityProofs, activeSet, nil
+}
+
+func prettyPrintProofs(proofs map[types.LayerID][]types.VotingEligibilityProof) string {
```
Comment (dshulyak) [inline miner/oracle.go:180]: this probably not the new code, but we should use zapcore.ObjectEncoder/zapcore.ArrayEncoder to log complex structs efficiently

**Review by dshulyak — APPROVED** [review]
hard to understand a bit what is new and what not, maybe git mv will be allow to track changes accurately in such cases

**Comment by countvonzero** [comment]
> hard to understand a bit what is new and what not, maybe git mv will be allow to track changes accurately in such cases

sorry about the diff. i did use `git mv` to begin with but maybe i made some git mistakes in the middle....

File: cmd/node/node.go (hunk 1 above)
Comment (countvonzero) [reply cmd/node/node.go]: changed

File: miner/oracle.go (hunk 2 above)
Comment (countvonzero) [reply miner/oracle.go:180]: not new code. i left TODO here to improve. thanks for the suggestion

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**